internal/services/managedapplications/managed_application_definition_resource_test.go

```diff
@@ -107,17 +107,17 @@ func TestAccManagedApplicationDefinition_update(t *testing.T) {
 }
 
 func (ManagedApplicationDefinitionResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
-	id, err := parse.ApplicationDefinitionID(state.ID)
+	id, err := applicationdefinitions.ParseApplicationDefinitionID(state.ID)
 	if err != nil {
 		return nil, err
 	}
 
-	resp, err := clients.ManagedApplication.ApplicationDefinitionClient.Get(ctx, id.ResourceGroup, id.Name)
+	resp, err := clients.ManagedApplication.ApplicationDefinitionClient.Get(ctx, *id)
 	if err != nil {
-		return nil, fmt.Errorf("retrieving Managed Definition %s (resource group: %s): %v", id.Name, id.ResourceGroup, err)
+		return nil, fmt.Errorf("retrieving Managed Definition %s : %v", id.String(), err)
```

Comment:
for consistency this can become:

```suggestion
		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
```